.github/actions/docusaurus-swizzled-warning/index.js

```diff
@@ -0,0 +1,135 @@
+'use strict'
+
+const core = require('@actions/core')
+const github = require('@actions/github')
+const fetch = require('node-fetch')
+const {
+  getAllFilesForPullRequest,
+  getChangedFilesBetweenTags,
+  findKeyEndingWith,
+} = require('./helpers')
+
+async function run() {
+  try {
+    const token = core.getInput('github-token', { required: true })
+
+    const { pull_request: pr } = github.context.payload
+    if (!pr) {
+      throw new Error('Event payload missing `pull_request`')
+    }
+
+    const client = github.getOctokit(token)
+    const packageName = 'docusaurus-theme-openapi'
+    const packageParentName = 'docusaurus-preset-openapi'
+    const overideComponents = ['Curl', 'Response']
+    const messageTemplate = `<table><thead><tr><th>
+      ⚠️ This PR contains changes to components of ${packageName} we've overridden
+    </th></tr></thead>
+    <tbody><tr><th>
+      We need to watch out for changes to the ${overideComponents.join(
+        ', ',
+      )}components
+    </th></tr>
+    `
+
+    if (
+      ['dependabot[bot]', 'dependabot-preview[bot]'].includes(pr.user.login)
+    ) {
+      const files = await getAllFilesForPullRequest(
+        client,
+        github.context.repo.owner,
+        github.context.repo.repo,
+        pr.number,
+      )
+
+      for (const file of files) {
+        if (file.filename !== 'package-lock.json') {
+          continue
+        }
```

Comment:
In general, it is harder to read large blocks of code that are deeply nested.

We can often work round nesting by taking some code like

```js
if (condition) {
  // do stuff
}
```

and re-writing it as

```js
if (!condition) {
  return
}

// do stuff
```

I think we could make this whole `run` function easier to read by getting rid of a couple of layers of nesting here. For example:

```js
if (!['dependabot[bot]', 'dependabot-preview[bot]'].includes(pr.user.login)) {
  return
}

const file = files.filter(f => f.filename === 'package-lock.json')[0]
if (file === undefined) {
   return
}

// do something with file
```

This would also make it clearer that we are not expecting to process more than one `package-lock.json`.

Reply:
I improved code readability with https://github.com/badges/shields/pull/9467/commits/81f020f69639a63f0c1a4a03b31c4add3b2db0a6
then fixed a type with https://github.com/badges/shields/pull/9467/commits/9e1c2e43b7f4be4f361c316b132359fa9bfb4439